src/utils/atomWithObservable.ts

```diff
@@ -120,3 +104,32 @@ export function atomWithObservable<TData>(
   )
   return observableAtom
 }
+
+function getInitialData<TData>(options: AtomWithObservableOptions<TData>) {
+  const initialData = options.initialData
+  return initialData instanceof Function ? initialData() : initialData
+}
+
+function firstValueFrom<T>(source: ObservableLike<T>): Promise<T> {
+  return new Promise<T>((resolve, reject) => {
+    let resolved = false
+    const subscription = source.subscribe({
```

Comment:
Although this looks cleaner, I think there's a risk of memory leaks.
Such case is, a) we create an atom, b) before getting the first value, c) atom can be unmounted.
We need to unsubscribe this subscription on `onMount` cleanup.

Reply:
@dai-shi hmm, I thought that the suspended component is not mounted before the promise resolves. So, we don't actually have an option to unsubscribe in any case or do I miss something?  

To illustrate, here is an atom that suspends and `onMount` is never called https://codesandbox.io/s/modern-glade-rutmog?file=/src/App.js:0-413

Reply:
BTW I've already changed other things. Should I do anything else here?

Reply:
Let's add a comment like this:
```js
// FIXME this implementation is not fully compatible with concurrent rendering.
// we need to deal with the case `onMount` is not invoked after the atom is initialized.

Reply:
@dai-shi I've added the comment. Please have a look